### PR TITLE
Add MSVC and macOS coverage

### PR DESCRIPTION
**Pull request by amyspark**
## Detailed description

This PR implements CI support for MSVC, Apple Xcode, and Homebrewed GCC for macOS.

For the latter two, dependencies are fetched from Homebrew as:

- `libftdi` depends on boost and others to be built from scratch
- GCC cannot build hidapi due to extensive usage of Objective-C constructs

This PR also includes the relevant fixes to make MSVC builds happy, in particular around standards compliance (#1742).

I'm open to any feedback. Let me know what you think!

## Your checklist for this pull request

* [X] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [X] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [X] It builds for hardware native (see [Building the firmware](https://github.com/blackmagic-debug/blackmagic?tab=readme-ov-file#building-black-magic-debug-firmware))
* [X] It builds as BMDA (see [Building the BMDA](https://github.com/blackmagic-debug/blackmagic?tab=readme-ov-file#building-black-magic-debug-app))
* [X] I've tested it to the best of my ability
* [X] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -152,6 +152,28 @@ if is_cross_build
 	)
 endif
 
+# Ensure that MSVC is switched to standards compliant mode
```
Comment (dragonmux) [inline meson.build:155]: Perhaps better to put this logic in `src/platforms/hosted/meson.build`? That way you can directly fix up `bmda_args` appropriately and use `cc`; It also puts the logic in with the rest of the Windows build logic.

File: src/target/riscv_debug.c [hunk 2]
```diff
@@ -523,7 +523,7 @@ static uint32_t riscv_hart_discover_isa(riscv_hart_s *const hart)
 	do {
 		DEBUG_INFO("Attempting %u-bit read on misa\n", hart->access_width);
 		/* Try reading the register on the guessed width */
-		uint32_t isa_data[4] = {};
+		uint32_t isa_data[4] = {0, 0, 0, 0};
```
Comment (dragonmux) [inline src/target/riscv_debug.c:526]: Ah, our bad! These are missing `U` on the end of each 0 to match the data type signedness, and you can drop all bar one leaving just `{0U}` if you want (C11 says the implied members are all 0's).

File: src/target/semihosting.c [hunk 3]
```diff
@@ -870,7 +870,7 @@ int32_t semihosting_request(target_s *const target, const uint32_t syscall, cons
 	target->tc->interrupted = false;
 
 	/* Set up the request block appropriately */
-	semihosting_s request = {r1, {}};
+	semihosting_s request = {r1, {0, 0, 0, 0}};
```
Comment (dragonmux) [inline src/target/semihosting.c:873]: Same comment as riscv_debug.c. The alternative patch (that works for us too) is:
```suggestion
	semihosting_s request = {r1};
```

**Review by dragonmux — CHANGES_REQUESTED** [review]
Only got a couple of notes as this all looks pretty great - with them addressed, this looks good to merge.

**Review by amyspark — COMMENTED** [review]
@dragonmux thanks for the comments! Bits should've been addressed now.

**Review by dragonmux — APPROVED** [review]
LGTM, merging. Thank you for the contribution!

**Pull request merged**